bundles/org.openhab.binding.tsmart/src/main/java/org/openhab/binding/tsmart/internal/TSmartHandler.java

```diff
@@ -0,0 +1,240 @@
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+
+/**
+ * The {@link TSmartHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author James Melville - Initial contribution
+ */
+@NonNullByDefault
+public class TSmartHandler extends BaseThingHandler {
+
+    private @Nullable TSmartConfiguration config;
+
+    private volatile int unRespondedRequests = 0;
+    private @Nullable InetAddress destAddr;
+
+    private @Nullable ScheduledFuture<?> statusRefreshJob;
+
+    public TSmartHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            requestStatusRefresh();
+        } else {
+            switch (channelUID.getId()) {
+                case CHANNEL_POWER:
+                    if (command instanceof OnOffType) {
+                        setPower(OnOffType.ON.equals(command));
+                    }
+                    break;
+                case CHANNEL_SETPOINT:
+                    if (command instanceof QuantityType<?> quantityCommand) {
+                        setSetpoint(quantityCommand);
+                    }
+                    break;
+                case CHANNEL_MODE:
+                    if (command instanceof StringType stringCommand) {
+                        setMode(stringCommand);
+                    }
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Initialise ThingHandler - start listening for UDP responses,
+     * and sending status requests on the configured frequency.
+     */
+    @Override
+    public void initialize() {
+        config = getConfigAs(TSmartConfiguration.class);
+        if (config.hostname.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        } else {
+            scheduler.schedule(this::startup, 0, TimeUnit.SECONDS);
+        }
+    }
+
+    private void startup() {
+        try {
+            InetAddress destination = InetAddress.getByName(config.hostname);
+
+            destAddr = destination;
+
+            TSmartUDPListener.addHandler(destination, this);
+
+            updateStatus(ThingStatus.UNKNOWN);
+
+            statusRefreshJob = scheduler.scheduleWithFixedDelay(this::requestStatusRefresh, 0, config.refreshInterval,
+                    TimeUnit.SECONDS);
+        } catch (UnknownHostException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
```

Comment:
In this case the binding won't be able to recover automatically. A real-life scenario could be after a power outage if openHAB is faster to start than the router, for example. See also #11891 which is almost exactly the same scenario. What I did here was to simply include the initialization into the scheduled job.